pkg/util/atomicfs/fsync.go

```diff
@@ -3,44 +3,70 @@
 package atomicfs
 
 import (
-	"bytes"
 	"io"
 	"os"
-	"path"
+	"path/filepath"
 
 	"github.com/grafana/dskit/multierror"
 )
 
-// CreateFile creates a file in the filePath, write the data into the file and then execute
-// fsync operation to make sure the file and its content are stored atomically. If the file already
-// exists, it will be overwritten.
-func CreateFile(filePath string, data io.Reader) error {
-	// Write the file, fsync it, then fsync the containing directory in order to guarantee
-	// it is persisted to disk. From https://man7.org/linux/man-pages/man2/fsync.2.html
-	//
-	// > Calling fsync() does not necessarily ensure that the entry in the
-	// > directory containing the file has also reached disk.  For that an
-	// > explicit fsync() on a file descriptor for the directory is also
-	// > needed.
```

Comment:
I think this comment on fsync'ing on a directory gives an important context. Otherwise, it wasn't clear for me why `File.Close` needs to do that.